src/main/java/org/apache/accumulo/testing/ingest/TestIngest.java

```diff
@@ -226,11 +227,8 @@ public static void ingest(AccumuloClient client, FileSystem fs, Opts opts,
     FileSKVWriter writer = null;
 
     if (opts.outputFile != null) {
-      writer = FileOperations
-          .getInstance()
-          .newWriterBuilder()
-          .forFile(opts.outputFile + "." + RFile.EXTENSION, fs, conf,
-              CryptoServiceFactory.newDefaultInstance())
+      writer = FileOperations.getInstance().newWriterBuilder()
```

Comment:
You could take this a little further.  There is a public API for creating Rfiles that could be used instead of FileOperations.  [RFile.newWriter()](https://static.javadoc.io/org.apache.accumulo/accumulo-core/2.0.0-alpha-2/org/apache/accumulo/core/client/rfile/RFile.html#newWriter--) is the entry point in the API for build an [RFileWriter](https://static.javadoc.io/org.apache.accumulo/accumulo-core/2.0.0-alpha-2/org/apache/accumulo/core/client/rfile/RFileWriter.html).  If this API is used then I think this class can avoid using FileSKVWriter, FileOperations, CryptoService, and possibly DefaultConfiguration.

Reply:
Yeah, I will take a look into that. Thanks for the idea.

Reply:
` RFileWriter writer = null;
    if (opts.outputFile != null) {
      writer = RFile.newWriter().to(opts.outputFile + ".rf").withFileSystem(fs).build();`
Would something along the lines of this work or am I missing something?

Reply:
For what it's worth, RFileWriter is AutoCloseable, and could be used inside a try-with-resources block. Should at least ensure it's closed in a finally block, if not null.

Reply:
`try(RfileWriter writer = RFile.newWriter().to(opts.outputFile + ".rf").withFileSystem(fs).build()){`
This is similar to the example given. Just wasn't sure if it was necessary. How does that look?

Reply:
Sorry on the formatting of the code, I saw you edited the last one

Reply:
> Sorry on the formatting of the code, I saw you edited the last one

No apologies necessary. I did that partly for readability, and partly as an demo/example for how to do it in future. It's not necessary to format things that way. It's just helpful once you learn to use all these little hidden gems in GitHub-flavored markdown. :smiley_cat:

Reply:
@ctubbsii What did you have in mind with the try-with block? Currently how the code is formatted, it would take some shifting of the rest of the function code to get everything needed inside the block. For example, the startTime and generateRow is used by both the RFile writer and batch writer. Also, there is a check to make sure to close the writer at the end of the function but I am not sure if that suffices or not.

Reply:
It was more of a passing thought... a suggestion to check. If the code is structured relatively linearly, the try-with-resources construct can be a clean and elegant way to keep the lifecycle of AutoCloseable resources, such as the RFile writer, well-managed. I didn't dive deep enough to look at the existing code structure, though.

Reply:
Yeah, I understand. It was a good suggestion and I will definitely keep it in mind for future use.

Reply:
@Manno15  that looks good.